### ISSUE TITLE
Lockscreen player controls not working

### ISSUE THREAD

**Issue by DrLuthor** [LeoNatan/LNPopupController#150]
#### Description

Hi! 
First of all: Thanks for the hard work!

So, here's my problem: when I'm on the lockscreen the play/pause, previous and next buttons do not work. It appears that  remoteControlReceivedWithEvent is not triggered.

#### Steps to Reproduce

remoteControlReceivedWithEvent is triggered if I go back to a classic pushed UIViewController.
Here's how I declared my LNPopupController
`PlayerViewController* player = [self.storyboard instantiateViewControllerWithIdentifier:@"Player"];        
        
      //load the document
        if (document.data_picture) {
            player.popupItem.image = document.data_picture;
        }
        else{
            player.popupItem.image = nil;
        }
        player.popupItem.title = document.name;
        if (document.author {
            player.popupItem.subtitle = document.author;
        }    
        player.popupItem.progress = 0.00;
    
        [self.tabBarController presentPopupBarWithContentViewController:player animated:YES completion:nil];
`
**EDIT**
I'm working with TextToSpeech API, not music (don't know if it's /specific to the issue).
So in my Player view controller, here's what I added as setup
`//init synthesizer
    _mainDelegate.synthesizerGlobal = [[AVSpeechSynthesizer alloc] init];
    _mainDelegate.synthesizerGlobal.delegate = self;
    
    //init to support lockscreen
    if ([self canBecomeFirstResponder]) {
        [self becomeFirstResponder];
    }
    [[UIApplication sharedApplication] beginReceivingRemoteControlEvents];`

#### Device, OS and Xcode Versions

iOS 10.2.1, Xcode 8.2.1, iPhone 5

**Comment by LeoNatan** [comment]
Interesting,
I have to say I am not familiar with this API. Is `remoteControlReceivedWithEvent:` supposed to be delivered on the view controller itself? I assume it works with the responder chain.

It might be related to #147. In the past, the popup content view controller was being added as a child controller to the presenting view controller, however I removed this due to issues with `UINavigationController` and `UITabBarController` making assumptions about their children and spazzing out if an extraneous controller is added there.

**Comment by DrLuthor** [comment]
**TLDR;** 
Solution : I moved remoteControlerReceivedWithEvent up the responder chain ie in the UIViewController which init my player

**Full story**
Well, I thought about something like this too. 
My first assumption was that the view controller wouldn't always be the first responder, mainly when the player is reduced to a bar.

So I tested the value [self canBecomeFirstResponder] which always returns false. But when I convert my player view controller back to a classic  UIViewController and tested it again, it appears that canBecomeFirstResponder retursn false either ... but the remoteControlReceivedWithEvet: is triggered.
The thing is, you cannot use remoteControlReceivedWithEvent in the appdelegate, which would have been great since my app use a tabbar. So I just had to duplicate it in each UIViewController (it sucks but I cannot subclass them).

Anyway, thanks for the time. It helped me solve my problem faster. I'd suggest you to mention this behaviour in your Readme.md since the lockscreen control is a nice feature for a music player app.

**Comment by LeoNatan** [comment]
Thanks for your report!
This is a workaround. I plan to fix this somehow. Started looking for solutions. For now, use the workaround.

**Comment by LeoNatan** [comment]
@DrLuthor Uhm, I got it working with no changes at all.
Did you implement support properly?

In my demo project, I changed it like so:

```objective-c
- (void)viewDidAppear:(BOOL)animated
{
	[super viewDidAppear:animated];
	
	[[UIApplication sharedApplication] beginReceivingRemoteControlEvents];
	[[AVAudioSession sharedInstance] setCategory: AVAudioSessionCategoryPlayback error:NULL];
	[[AVAudioSession sharedInstance] setActive:YES error:NULL];
	
	_player = [[AVAudioPlayer alloc] initWithContentsOfURL:[[NSBundle mainBundle] URLForResource:@"music" withExtension:@"m4a"] error:NULL];
	[_player prepareToPlay];
	[_player play];
	
	[self becomeFirstResponder];
}

- (BOOL)canBecomeFirstResponder
{
	return YES;
}

- (void)remoteControlReceivedWithEvent:(UIEvent *)event
{
	NSLog(@“%@“, event);
}
```

`remoteControlReceivedWithEvent:` is called correctly. However, this is less then optimal, because if you close the popup, it loses first responder status. This is the same with popping a view controller.

However, placing `remoteControlReceivedWithEvent:` in AppDelegate gets called correctly in my demo app, I think the issue is on your side. The AppDelegate is part of the responder chain.

I have pushed a branch that demonstrates this behavior and closing the issue for now. Let me know if you have questions.
